sdk/core/core-tracing/src/createSpan.ts

```diff
@@ -48,6 +64,78 @@ export function isTracingDisabled(): boolean {
   return Boolean(azureTracingDisabledValue);
 }
 
+/**
+ * Maintains backwards compatibility with the previous `OperationTracingOptions` in core-tracing preview.13 and earlier
+ * which passed `spanOptions` as part of `tracingOptions`.
+ */
+function disambiguateParameters<T extends { tracingOptions?: OperationTracingOptions }>(
+  operationOptions: T,
+  startSpanOptions?: SpanOptions
+): [OperationTracingOptions, SpanOptions, T] {
+  const { tracingOptions } = operationOptions;
+
+  // If startSpanOptions is provided, then we are using the new signature,
+  // otherwise try to pluck it out of the tracingOptions.
+  const spanOptions: SpanOptions = startSpanOptions || (tracingOptions as any)?.spanOptions || {};
+  spanOptions.kind = spanOptions.kind || SpanKind.INTERNAL;
+
+  return [tracingOptions || {}, spanOptions, operationOptions];
```

Comment:
Why do we return `operationOptions` here? We're returning the same object reference that was passed in, so the caller already has it.